scripts/global.gd

```diff
@@ -40,11 +45,15 @@ func setup_timer(time_limit: int):
 	timer.one_shot = true
 	timer.timeout.connect(_on_timer_timeout)
 	add_child(timer)
-	timer.start(time_limit)
-	timer_added.emit()
+	timer_time_limit = time_limit
+	#timer.start(time_limit)
+	#timer_added.emit()
 
 
 func _on_timer_timeout():
+	#If the game is already ended, do nothing.
+	if is_ended:
+		return
```

Comment:
I don't think this is strictly necessary because you stop the timer when the game ends; though it doesn't hurt of course.